common/lib/xmodule/xmodule/html_module.py

```diff
@@ -253,6 +254,24 @@ def non_editable_metadata_fields(self):
         non_editable_fields.append(HtmlDescriptor.use_latex_compiler)
         return non_editable_fields
 
+    def index_dictionary(self):
+        xblock_body = super(HtmlDescriptor, self).index_dictionary()
+        html_content = re.sub(r"(\s|&nbsp;|//)+", " ", html_to_text(self.data))
+        html_content = re.sub(r"<!\[CDATA\[.*\]\]>", "", html_content)
+        html_content = re.sub(r"<!--.*-->", "", html_content)
```

Comment:
Can you add some comments to these lines explaining what they are doing and why? Not everyone will immediately recognize what these strings mean. Could be as simple as `# Removing HTML-encoded non-breaking space characters`, `# removing HTML CDATA`, and `# removing HTML comments`. Also, why are you specifically removing these strings, and no others? What's special about these three types of content?